apps/voting/contracts/Voting.sol

```diff
@@ -213,6 +240,72 @@ contract Voting is IForwarder, AragonApp {
         _executeVote(_voteId);
     }
 
+    /**
+    * @notice Assign `_delegate` as the delegate for the sender
+    * @param _delegate address to delegate to
+    */
+    function setDelegate(address _delegate) external {
+        require(_delegate != address(0), ERROR_ZERO_ADDRESS_PASSED);
+        require(_delegate != msg.sender, ERROR_SELF_DELEGATE);
+
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(_delegate != prevDelegate, ERROR_DELEGATE_SAME_AS_PREV);
+
+        uint256 votingPower = token.balanceOfAt(msg.sender, getBlockNumber64() - 1);
+        require(votingPower > 0, ERROR_NO_VOTING_POWER);
+
+        if (prevDelegate != address(0)) {
+            _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        }
+        _addDelegatedAddressFor(_delegate, msg.sender);
+    }
+
+    /**
+    * @notice Unassign `_delegate` from the sender
+    */
+    function resetDelegate() external {
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(prevDelegate != address(0), ERROR_DELEGATE_NOT_SET);
+
+        _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        delete delegates[msg.sender];
+    }
+
+    /**
+     * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of the `_voters` list
+     * @param _voteId Id for vote
+     * @param _supports Whether the delegate supports the vote
+     * @param _voters list of voters
+     */
+    function attemptVoteForMultiple(uint256 _voteId, bool _supports, address[] _voters) public voteExists(_voteId) {
+        Vote storage vote_ = votes[_voteId];
+        require(_isValidPhaseToVote(vote_, _supports), ERROR_CAN_NOT_VOTE);
+        bool hasManagedToVote = false;
+
+        for (uint256 i = 0; i < _voters.length; ++i) {
+            address voter = _voters[i];
+            require(_hasVotingPower(vote_, voter), ERROR_NO_VOTING_POWER);
```

Comment:
seems we can use the `skip` logic also for voting power
i.e. use `if (_canVoteFor(vote_, msg.sender, voter) && _hasVotingPower(vote_, voter)) {...` instead of `require`

will it be more UX-oriented in terms of operational costs?

Reply:
voting power is fixed on snapshot and cannot change, so, it looks ok. Anyway, it's a bad idea to shove random addresses to this method.

Reply:
I would really want to see a comment with description why we are mixing `require` approach with `skip_on_failure`, because it's not really obvious why so